File: php/src/RequestOptions.php
<?php

declare(strict_types=1);

namespace Buble;

final class RequestOptions
{
    /**
     * @param array<string, string> $headers
     * @param array<string, string> $query
     */
    public function __construct(
        public readonly array $headers = [],
        public readonly array $query = [],
    ) {
    }
}

File: dotnet/src/Buble.Sdk/RequestOptions.cs
namespace Buble.Sdk;

/// <summary>
/// Optional headers and query string values for a single request.
/// </summary>
public sealed class RequestOptions
{
    public IDictionary<string, string> Headers { get; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);

    public IDictionary<string, string> Query { get; } = new Dictionary<string, string>(StringComparer.Ordinal);

    public static RequestOptions None => new();
}
